app/login.tsx

```diff
@@ -0,0 +1,348 @@
+                <TextInput
+                  style={styles.input}
+                  placeholder='E-mail'
+                  placeholderTextColor='#8E8E93'
+                  value={value}
+                  onChangeText={onChange}
+                  onBlur={onBlur}
+                  keyboardType='email-address'
+                  autoCapitalize='none'
+                  autoCorrect={false}
+                />
+              )}
+            />
+            {errors.email && (
+              <Text style={styles.errorText}>{errors.email.message}</Text>
+            )}
+          </View>
+
+          {/* Password Input */}
+          <View style={styles.inputContainer}>
+            <View style={styles.passwordContainer}>
+              <Controller
+                control={control}
+                name='password'
+                render={({ field: { onChange, onBlur, value } }) => (
+                  <TextInput
+                    style={styles.input}
+                    placeholder='Password'
+                    placeholderTextColor='#8E8E93'
+                    value={value}
+                    onChangeText={onChange}
+                    onBlur={onBlur}
+                    secureTextEntry={!showPassword}
+                    autoCapitalize='none'
+                    autoCorrect={false}
+                  />
+                )}
+              />
+              <Pressable
+                style={styles.eyeButton}
+                onPress={() => setShowPassword(!showPassword)}
+              >
+                <Ionicons
+                  name={showPassword ? 'eye-off' : 'eye'}
+                  size={20}
+                  color='#8E8E93'
+                />
+              </Pressable>
+            </View>
+            {errors.password && (
+              <Text style={styles.errorText}>{errors.password.message}</Text>
+            )}
+          </View>
+
+          {/* Forgot Password */}
+          <Pressable
+            style={styles.forgotPasswordContainer}
+            onPress={handleForgotPassword}
+          >
+            <Text style={styles.forgotPasswordText}>Forgot password?</Text>
+          </Pressable>
+
+          {/* Sign In Button */}
+          <TouchableOpacity
+            style={[
+              styles.signInButton,
+              isLoading && styles.signInButtonDisabled,
+            ]}
+            onPress={handleSubmit(onSubmit)}
+            disabled={isLoading}
+          >
+            <Text style={styles.signInButtonText}>
+              {isLoading ? 'Signing in...' : 'Sign in'}
+            </Text>
+          </TouchableOpacity>
+
+          {/* Or Divider */}
+          <View style={styles.dividerContainer}>
+            <View style={styles.dividerLine} />
+            <Text style={styles.dividerText}>or</Text>
+            <View style={styles.dividerLine} />
+          </View>
+
+          {/* Social Login Buttons */}
+          <View style={styles.socialButtonsContainer}>
+            <TouchableOpacity
```

Comment:
All your social buttons are sharing the same styles, and also has the same format, so this is a good candidate for a component. Create a component called SocialButton